src/Particle/SoaDistanceTableAAOMPTarget.h

```diff
@@ -198,10 +198,13 @@ struct SoaDistanceTableAAOMPTarget : public DTD_BConds<T, D, SC>, public Distanc
   {
     ScopedTimer local_timer(move_timer_);
 
-    old_prepared_elec_id = prepare_old ? iat : -1;
     temp_r_.attachReference(temp_r_mem_.data(), temp_r_mem_.size());
     temp_dr_.attachReference(temp_dr_mem_.size(), temp_dr_mem_.capacity(), temp_dr_mem_.data());
 
+    // should this really be P.activePtcl() or iat?
+    // The activePtcl state value in ParticleSet seems like a landmine. When you get here from makeVirtualMoves_
+    // P.activePtl() = -1 but it was that iat  == 0 and prepare_old = true.  This as far as I can tell
```

Comment:
When P.activePtcl or iat is -1, should always be false. It is worth adding an assertion.
assert((prepare_old && iat >=0 && iat < N_targets) || !prepare_old)

Reply:
adding this